Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchyBasedLinkResolver.swift

```diff
@@ -58,6 +57,42 @@ final class PathHierarchyBasedLinkResolver {
         }
     }
     
+    /// Returns the direct descendants of the given page that match the given source language filter.
+    ///
+    /// A descendant is included if it has a language representation in at least one of the languages in the given language filter or if the language filter is empty.
+    ///
+    /// - Parameters:
+    ///   - reference: The identifier of the page whose descendants to return.
+    ///   - languagesFilter: A set of source languages to filter descendants against.
+    /// - Returns: The references of each direct descendant that has a language representation in at least one of the given languages.
+    func directDescendants(of reference: ResolvedTopicReference, languagesFilter: Set<SourceLanguage>) -> Set<ResolvedTopicReference> {
+        guard let id = resolvedReferenceMap[reference] else { return [] }
+        let node = pathHierarchy.lookup[id]!
+        
+        func directDescendants(of node: PathHierarchy.Node) -> [ResolvedTopicReference] {
```

Comment:
This whole function might fit better as an instance method of PathHierarchy.Node. And you could even pass in the language filter. So you could write something like this below?

```
results.formUnion(node.directDescendants(languages: languagesFilter))
```

Reply:
I'll think about it. I personally try make private single use API first and then consolidate it when there are multiple places that have similar needs. 

There is also _some_ overlap between this and `PathHierarchy.DisambiguationContainer.disambiguatedValues(includeLanguage:)` but I don't have a clear picture yet of what could be a unifying API there. 

Luckily it's all internal and isolated API so it's easy to change.